crates/sqlexec/src/planner/physical_plan/drop_tables.rs

```diff
@@ -101,16 +105,20 @@ async fn drop_tables(
         })
     });
 
+    // we want to make sure that the catalog is updated before we delete the delta tables
     mutator
         .mutate(plan.catalog_version, drops)
         .await
         .map_err(|e| DataFusionError::Execution(format!("failed to drop tables: {e}")))?;
 
-    // // Run background jobs _after_ tables get removed from the catalog.
-    // //
-    // // TODO: If/when we have transactions, background jobs should be stored
-    // // on the session until transaction commit.
-    // self.background_jobs.add_many(jobs)?;
+    // only after the catalog is updated, we can delete the delta tables
+    // TODO: this should be done in the scheduler.
+    let sys_exec =
+        SystemOperationExec::new(DeleteDeltaTablesOperation::new(plan.tbl_entries.clone()).into());
+    let _ = sys_exec
+        .execute(0, context.clone())?
+        .collect::<Vec<_>>()
+        .await;
```

Comment:
what are the items in the iterator that we're throwing away? 

The first time I read this it seemed like we were throwing away an error, and now I'm unsure.

Reply:
We just discard the stream of recordbatches. The error is still propagated if the execution fails, but we don't care about the results.